### ISSUE TITLE
Protocol cannot distinguish empty string from null string.

### ISSUE THREAD

**Issue by hvbtup** [nanosai/rion-ops-java#14]
I think this is worth mentioning:

The protocol treats a value as null if the length-length is null.
Serializing the two Java String objects null and "" (empty string) results in the same serialized form.

Thus when deserializing the input "" becomes null.
This can be an issue in some cases.

We stumbled upon this because after changing our serialization code for UI properties, some labels which should be empty showed the null-replacement text "(not set)".

**Comment by jjenkov** [comment]
Strange. That is a bug. It should serialize an empty string "" to 51 00   (UTF-8, 1 length byte with the value 0). Not to 50  (UTF-8, 0 length bytes = null).

I will look at it this week and fix it.

**Comment by hvbtup** [comment]
This is probably because in my serialization code I am using UTF_8_SHORT in case the string length is <=15.
So the workaround is to *not* use UTF_8_SHORT if the string is empty?

**Comment by hvbtup** [comment]
Maybe it is enough to change the if-statement in the several writeUtf8 methods like this:

OLD: if(length <= 15){
NEW: if(length > 0 && length <= 15){

**Comment by jjenkov** [comment]
I will make a new release of RION Ops with better null support. It should also be possible to write null Int64 fields, Boolean fields, Calendar fields, Object fields etc. I will add methods for that. And then I will make sure that existing methods support null properly.

**Comment by jjenkov** [comment]
This new release is now in progress...

**Comment by jjenkov** [comment]
I seem to have reproduced the error now in a unit test (RionReaderTest.testWriteReadNull() )

**Comment by jjenkov** [comment]
Your suggested fix seems to have fixed the error. An empty string is now serialized as a UTF-8 field with 1 length byte with the value 0 (2 bytes in total). Thus, it is now read correctly as an empty string by RionReader.readutf8String() .

**Comment by jjenkov** [comment]
This bug has now been fixed in RionWriter.writeUtf8(String) in v. 0.9.0 which was released this evening.

I've noticed that the same error exists in 

RionWriter.writeUtf8( byte[] ) 
RionWriter.writeUtf8(byte[], int offset, int length)

RionFieldWriterString.writeValueField()

I will fix those in a v. 0.9.1 over the following days - but at least now RionWriter.writeUtf8() works, and I know what the problem is, and what fix is for the rest of the places the error exists.
